src/control/cmd/agent/mgmt_rpc.go

```diff
@@ -47,13 +59,36 @@ type mgmtModule struct {
 	ap               string
 	tcfg             *security.TransportConfig
 	cachedAttachInfo bool
-	resmgmtpb        []byte
+	// maps NUMA affinity and device index to a response
+	resmgmtpb map[uint]map[int][]byte
+	// maps NUMA affinity to a device index
+	devIdx     map[uint]int
+	clientData map[int32]clientProcessCfg
+	mutex      sync.Mutex
```

Comment:
While you're reworking this a bit, I think it would be cleaner to extract this caching functionality into a separate structure with its own methods and unit tests, rather than bolting more stuff onto the mgmtModule struct. Doing it that way will make it easier to modify the agent code in a more decoupled way. For example, we're going to need to have a way to dump the attachinfo data to a file in order to complete [DAOS-4392](https://jira.hpdd.intel.com/browse/DAOS-4392). Not suggesting that you need to handle that work in this PR, but would just like it to be kept in mind so that we don't make life harder for ourselves later.

Reply:
Ok, I gave a crack at the restructuring.  See if that's what you had in mind.  If so, I'll go ahead with making some unit tests.